xds/src/main/java/io/grpc/xds/GcpAuthenticationFilter.java

```diff
@@ -134,31 +135,59 @@ public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
           MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
 
         String clusterName = callOptions.getOption(CLUSTER_SELECTION_KEY);
+        if (clusterName == null) {
+          return new FailingClientCall<>(
+              Status.UNAVAILABLE.withDescription(
+                  String.format(
+                      "GCP Authn for %s does not contain cluster resource", filterInstanceName)));
+        }
+
         if (!clusterName.startsWith("cluster:")) {
           return next.newCall(method, callOptions);
         }
         XdsConfig xdsConfig = callOptions.getOption(XDS_CONFIG_CALL_OPTION_KEY);
+        if (xdsConfig == null) {
+          return new FailingClientCall<>(
+              Status.UNAVAILABLE.withDescription(
+                  String.format(
+                      "GCP Authn for %s with %s does not contain xds configuration",
+                      filterInstanceName, clusterName)));
+        }
+
         StatusOr<XdsClusterConfig> xdsCluster =
-            xdsConfig.getClusters().get(clusterName);
+            xdsConfig.getClusters().get(clusterName.substring(8)); // get rid of prefix "cluster:"
+        if (xdsCluster == null) {
+          return new FailingClientCall<>(
+              Status.UNAVAILABLE.withDescription(
+                  String.format(
+                      "GCP Authn for %s with %s does not contain xds cluster",
+                      filterInstanceName, clusterName)));
+        }
+
         if (!xdsCluster.hasValue()) {
           return next.newCall(method, callOptions);
         }
-        CdsUpdate cdsUpdate = xdsCluster.getValue().getClusterResource();
-        if (cdsUpdate == null) {
-          return new FailingClientCall<>(
-              Status.UNAVAILABLE.withDescription("CDS resource unavailable"));
-        }
+        ImmutableMap<String, Object> parsedMetadata = xdsCluster.getValue().getClusterResource()
+            .parsedMetadata();
 
-        if (!cdsUpdate.parsedMetadata().containsKey("FILTER_INSTANCE_NAME")) {
+        if (parsedMetadata == null || !parsedMetadata.containsKey(filterInstanceName)) {
           return next.newCall(method, callOptions);
         }
 
         AudienceWrapper audience;
-        try {
-          audience = (AudienceWrapper) cdsUpdate.parsedMetadata().get(filterInstanceName);
-        } catch (ClassCastException e) {
+        if (parsedMetadata.get(filterInstanceName) instanceof AudienceWrapper) {
+          audience = (AudienceWrapper) parsedMetadata.get(filterInstanceName);
+          if (audience.audience == null) {
```

Comment:
How can `audience.audience` be null? I'd be happy for you to add a checkNotNull to make it more obvious, but the config should have been rejected instead of getting a null here.